pkg/codegen/testing/test/testdata/simple-enum-schema/go/plant/x/pulumiTypes.go

```diff
@@ -0,0 +1,109 @@
+// Code generated by test DO NOT EDIT.
+// *** WARNING: Do not edit by hand unless you're certain you know what you are doing! ***
+
+package plant
+
+import (
+	"context"
+	"reflect"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumix"
+	"simple-enum-schema/plant/internal"
+)
+
+var _ = internal.GetEnvOrDefault
+
+type Container struct {
+	Brightness *ContainerBrightness `pulumi:"brightness"`
+	Color      *string              `pulumi:"color"`
+	Material   *string              `pulumi:"material"`
+	Size       ContainerSize        `pulumi:"size"`
+}
+
+// Defaults sets the appropriate defaults for Container
+func (val *Container) Defaults() *Container {
+	if val == nil {
+		return nil
+	}
+	tmp := *val
+	if tmp.Brightness == nil {
+		brightness_ := ContainerBrightness(1.0)
+		tmp.Brightness = &brightness_
+	}
+	return &tmp
+}
+
+type ContainerArgs struct {
+	Brightness pulumix.Input[*ContainerBrightness] `pulumi:"brightness"`
+	Color      pulumix.Input[*string]              `pulumi:"color"`
+	Material   pulumix.Input[*string]              `pulumi:"material"`
+	Size       pulumix.Input[ContainerSize]        `pulumi:"size"`
+}
+
+// Defaults sets the appropriate defaults for ContainerArgs
+func (val *ContainerArgs) Defaults() *ContainerArgs {
+	if val == nil {
+		return nil
+	}
+	tmp := *val
+	if tmp.Brightness == nil {
+		tmp.Brightness = pulumix.Ptr(ContainerBrightness(1.0))
+	}
+	return &tmp
+}
+func (ContainerArgs) ElementType() reflect.Type {
+	return reflect.TypeOf((*Container)(nil)).Elem()
+}
+
+func (i ContainerArgs) ToContainerOutput() ContainerOutput {
+	return i.ToContainerOutputWithContext(context.Background())
+}
+
+func (i ContainerArgs) ToContainerOutputWithContext(ctx context.Context) ContainerOutput {
+	return pulumi.ToOutputWithContext(ctx, i).(ContainerOutput)
+}
+
+func (i *ContainerArgs) ToOutput(ctx context.Context) pulumix.Output[*ContainerArgs] {
+	return pulumix.Val(i)
+}
+
+type ContainerOutput struct{ *pulumi.OutputState }
+
+func (ContainerOutput) ElementType() reflect.Type {
+	return reflect.TypeOf((*Container)(nil)).Elem()
+}
+
+func (o ContainerOutput) ToContainerOutput() ContainerOutput {
+	return o
+}
+
+func (o ContainerOutput) ToContainerOutputWithContext(ctx context.Context) ContainerOutput {
+	return o
+}
+
+func (o ContainerOutput) ToOutput(ctx context.Context) pulumix.Output[Container] {
+	return pulumix.Output[Container]{
+		OutputState: o.OutputState,
+	}
+}
```

Comment:
It seems like something fishy is going on here. Let's not block on it for this PR, but would like to figure it out subsequently.

1. For the generic variants, I'm not sure there's a need for `ToFooOutput` and `ToFooOutputWithContext` methods.
2. As I mentioned elsewhere, I think we can use a type definition for the Output type, which I think avoids the need to generate `ElementType` and `ToOutput` methods.
3. `(i *ContainerArgs) ToOutput(ctx context.Context) pulumix.Output[*ContainerArgs]` looks suspect to me. I understand why it's that way at the moment to get things to work, but I think we need to revisit this. It seems like the signature should be something like:

    ```go
    (i *ContainerArgs) ToOutput(ctx context.Context) pulumix.Output[Container]
    ```

    or:

    ```go
    (i ContainerArgs) ToOutput(ctx context.Context) pulumix.Output[Container]
    ```


```suggestion
func (i *ContainerArgs) ToOutput(ctx context.Context) pulumix.Output[*ContainerArgs] {
	return pulumix.Val(i)
}

type ContainerOutput pulumix.Output[Container]
```